agimus_controller/agimus_controller/warm_start_reference.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+import numpy.typing as npt
+
+import pinocchio as pin
+
+from agimus_controller.warm_start_base import WarmStartBase
+from agimus_controller.trajectory import TrajectoryPoint
+
+
+class WarmStartReference(WarmStartBase):
+    def __init__(self) -> None:
+        super().__init__()
+        self._rmodel: pin.Model | None = None
+        self._rdata: pin.Data | None = None
+
+    def setup(self, rmodel: pin.Model) -> None:
+        self._rmodel = rmodel
+        self._rdata = self._rmodel.createData()
+
+    def generate(
+        self,
+        initial_state: TrajectoryPoint,
+        reference_trajectory: list[TrajectoryPoint],
+    ) -> tuple[
+        npt.NDArray[np.float64],
+        list[npt.NDArray[np.float64]],
+        list[npt.NDArray[np.float64]],
+    ]:
+        """
```

Comment:
I started to dig into the documentation and for overriding function we could document it [like this](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#3831-overridden-methods)

Reply:
how would you do it? Have the same docstrings? I'm slightly worried that abstract class should be more high-level, I wouldn't mention inverse dynamics there